images/Images/FITSImage.h

```diff
@@ -219,8 +219,11 @@ class FITSImage: public ImageInterface<Float>
   // Check class invariants.
   virtual Bool ok() const;
 
-  // Return the (internal) data type (TpFloat or TpShort).
-  DataType dataType () const
+  // Return the data type (TpFloat).
+  virtual DataType dataType() const;
+
+  // Return the (internal) data type.
+  DataType internalDataType() const
```

Comment:
this sounds like it should be private or it would be good to have some documentation what the difference is to the datatype